Thoughtless-Eels/Bangazon-WebApp
# Gl adding models to db context, setting up database & first migration

## Link to Ticket
Ticket: No direct ticket reference

## Description of Proposed Changes

- Finalized models
- added photo url string property to product model
- Added database connection
- Performed initial migration
- edited the .gitignore


## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout gl-addingModelsToDbContext
dotnet restore
```
Open the Bangazon.sln file
Open Package Manager Console by clicking Tools in the top menu bar then click Nugget Package Manager, then choose Package Manager Console.
Type: 
```
Update-Database
```
and press enter

Check that there are no errors when the Update-Database function completes. 


## Impacted Areas in Application

List general components of the application that this PR will affect:

* Product Model
* ApplicationDbContext
* Migration folder


## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
